hw/ip/rom_ctrl/util/mem.py

```diff
@@ -85,9 +66,7 @@ def add_ecc32(self) -> None:
         bits, to make 39-bit words.
 
         '''
-        codes = secded_gen.gen_code('hsiao', 32, 7)
-        bitmasks = secded_gen.calc_bitmasks(32, 7, codes, False)
-        self.words = [add_ecc32(w, bitmasks) for w in self.words]
+        self.word = [ecc_encode('inv_hsiao', 32, w) for w in self.words]
```

Comment:
This should probably be `self.words`

Reply:
I think that maybe we should do https://github.com/lowRISC/opentitan/pull/9607 first, after which this patch will just be adding an "inv_" prefix in this file.

Reply:
yes indeed, it should :).

Anyway, thanks for the cleanup.

I rebased this onto your change, and split out the OTP changes into #9472, since these are unrelated to the bus-integrity scheme.

I hope I can take another stab at the FPGA failures later today (will try what you have suggested, but I guess at some point an in-system debug will be necessary).

Reply:
My other thought was that you could change the host code to spit out the values of INSN_COUNT and ERR_BITS. Assuming that INSN_COUNT is small (I bet it is!) you can manually trace through the OTBN code (run it in the standalone simulator, for example) to see which instruction is causing the fault.

Reply:
Thanks @rswarbrick - @vogelpi and I figured it out via a couple of synthesis experiments and it finally works now!